conans/test/functional/profiles/test_python_profiles.py

```diff
@@ -0,0 +1,217 @@
+import os
+import textwrap
+from collections import defaultdict
+from importlib.util import spec_from_file_location, module_from_spec
+
+import pytest
+
+from conans.client.tools import environment_append
+from conans.test.utils.tools import TestClient
+from conans import ProfileSettings, ProfileOptions, ProfileEnv
+
+
+class ProfileLoader(object):
+
+    @staticmethod
+    def _init_profile():
+        import conans
+        conans.settings = ProfileSettings()
+        conans.options = ProfileOptions()
+        conans.env = ProfileEnv(list)
+
+    @staticmethod
+    def _load_module_from_file(filename):
+        name = os.path.splitext(os.path.basename(filename))[0]
+        spec = spec_from_file_location(name, filename)
+        config = module_from_spec(spec)
+        spec.loader.exec_module(config)
+        return config
+
+    @staticmethod
+    def _load_profile(profile_path):
+        profile = {}
+        profile_content = ProfileLoader._load_module_from_file(profile_path)
+        for key, value in profile_content.__dict__.items():
+            if not key.startswith("__"):
+                profile[key] = value
+        return profile.copy()
+
+    @staticmethod
+    def load_profile_python(profile_path, parent_profile_path=None):
+        # TODO: implement include() of other profiles? variable substitution?
+        if parent_profile_path:
+            ProfileLoader._load_profile(parent_profile_path)
+        profile = ProfileLoader._load_profile(profile_path)
+        ProfileLoader._init_profile()
+        return profile.copy()
+
+
+@pytest.fixture
+def client():
+    return TestClient()
```

Comment:
A fixture of 1 line provides no value, they are for initialization reuse, but 1 liner is not worth (actually counter productive, losing locality is bad for reading tests). You can put all 3 fixtures in just 1, if you are always using them together: ``(client, pr1_content, pr2_content):``. Also include the ``client.save()`` in the fixture.